packages/fluentui/react-northstar/src/components/Status/Status.tsx

```diff
@@ -36,6 +36,65 @@ export interface StatusProps extends UIComponentProps {
 export type StatusStylesProps = Pick<StatusProps, 'color' | 'size' | 'state'>;
 export const statusClassName = 'ui-status';
 
+const useStatusStyles = makeStyles([
+  [
+    null,
+    tokens => ({
+      alignItems: 'center',
+      display: 'inline-flex',
+      justifyContent: 'center',
+      verticalAlign: 'middle',
+
+      borderRadius: '9999px',
+
+      backgroundColor: tokens.colorScheme.default.background5,
+    }),
+  ],
+
+  [{ state: 'success' }, tokens => ({ backgroundColor: tokens.colorScheme.green.background })],
```

Comment:
I'm not exactly sure how this will work in extraction. Here's a scenario:

```jsx
makeStyles([
  [ null, { background: 'red' } ],
  [ state: 'info' , { background: 'green' } ]
]);
```

In this case, we have 2 possible atomic css classes (one for each bg color.) Lets call these classes `r` and `g`.

If state is not info, the className should default to `r`. If the state is `info`, it should default to `g`.

My initial extraction thought was that this would be extracted to predictable hash classes: `r-1` and `g-2` for example purposes:

```jsx
// extract this stylesheet: `.r-1{background:red}.g-2{background:green}`.

// code is transformed into this:
makeStyles([
[ null, 'r-1' ],
[ state: 'info' , 'g-2' ]
]);
```
...but that's not right! I believe with the current logic, when state is `info`, this would equate to the className `r-1 g-2` which might be green or red depending on load order.

So I think the question I'd like to more clearly understand is how this should get translated in the extraction.

I've prototyped an initial extractor here:
https://codesandbox.io/s/makestylesextractor-prototype-owk84?file=/src/makeStylesExtractor.test.js:1416-1506

Reply:
> But I believe that means that we will also have to inject some javascript for rehydrating the classname hash map. Without priming the hash map, the class names will be joined, resulting in specificity to control the result.
>
> Am I understanding this correctly?

A small correction, it will keep property name and we will perform a merge later based on it:

```jsx
makeStyles([
  [null, { background: "r" }],
  [{ primary: true }, { background: "b" }],
  [{ state: "info" }, { background: "g" }]
]);
```

So for `{ primary: true }` we will get:

```js
[{ background: "r" }, { background: "b" }];
```

Once we will shallow merge:

```js
const classes = Object.values(
  Object.assign({}, { background: "r" }, { background: "b" })
); // ["b"]
```